hack/generator/pkg/codegen/pipeline/add_cross_resource_references.go

```diff
@@ -278,5 +278,38 @@ func newKnownReferencesMap(configuration *config.Configuration) map[referencePai
 			typeName: astmodel.MakeTypeName(configuration.MakeLocalPackageReference("microsoft.compute", "v1alpha1api20200930"), "DiskProperties"),
 			propName: "DiskAccessId",
 		}: true,
+		// AKS
+		{
+			typeName: astmodel.MakeTypeName(configuration.MakeLocalPackageReference("microsoft.containerservice", "v1alpha1api20210501"), "PrivateLinkResource"),
+			propName: "Id",
+		}: true,
+		{
+			typeName: astmodel.MakeTypeName(configuration.MakeLocalPackageReference("microsoft.containerservice", "v1alpha1api20210501"), "ManagedClusterAgentPoolProfile"),
+			propName: "VnetSubnetID",
+		}: true,
+		{
+			typeName: astmodel.MakeTypeName(configuration.MakeLocalPackageReference("microsoft.containerservice", "v1alpha1api20210501"), "ManagedClusterAgentPoolProfile"),
+			propName: "PodSubnetID",
+		}: true,
+		{
+			typeName: astmodel.MakeTypeName(configuration.MakeLocalPackageReference("microsoft.containerservice", "v1alpha1api20210501"), "ManagedClusterAgentPoolProfileProperties"),
+			propName: "VnetSubnetID",
+		}: true,
+		{
+			typeName: astmodel.MakeTypeName(configuration.MakeLocalPackageReference("microsoft.containerservice", "v1alpha1api20210501"), "ManagedClusterAgentPoolProfileProperties"),
+			propName: "PodSubnetID",
+		}: true,
+		{
+			typeName: astmodel.MakeTypeName(configuration.MakeLocalPackageReference("microsoft.containerservice", "v1alpha1api20210501"), "UserAssignedIdentity"),
+			propName: "ResourceId",
+		}: true,
+		{
+			typeName: astmodel.MakeTypeName(configuration.MakeLocalPackageReference("microsoft.containerservice", "v1alpha1api20210501"), "ResourceReference"),
+			propName: "Id",
+		}: true,
+		{
+			typeName: astmodel.MakeTypeName(configuration.MakeLocalPackageReference("microsoft.containerservice", "v1alpha1api20210501"), "Componentsqit0Etschemasmanagedclusterpropertiespropertiesidentityprofileadditionalproperties"),
```

Comment:
What the heck? I guess this is one of those names that need a bit of love.

Reply:
So actually I think in some of our SDKs this is possibly called this as well. This is the actual type name in Swagger I think...

Reply:
Ouch